src/ITensorChainRules/zygoterules.jl

```diff
@@ -0,0 +1,13 @@
+
+# Needed for defining the rule for `adjoint(A::ITensor)`
+# which currently doesn't work by overloading `ChainRulesCore.rrule`
+using ZygoteRules: @adjoint
+
+@adjoint function Base.adjoint(x::ITensor)
+  y = prime(x)
+  function setinds_pullback(ȳ)
```

Comment:
I have an implementation https://github.com/LinjianMa/PEPSAD.jl/blob/master/src/peps.jl#L125 that uses a map to check if the index we set is correct or not. I can take this if it sounds reasonable

Reply:
Yes I didn't consider the case where the indices might be permuted, good point.

I think going through each prime/tag operation and defining the pullback in terms of the "inverse" operation is sufficient, such as `prime(A) -> prime(A, -1)` or `addtags(A, "x") -> removetags(A, "x")`. Note that `noprime` is not always the reverse of `prime`, since you might prime from level 1 to 2.

Reply:
makes sense

Reply:
Should be fixed in the latest PR. I define a function `inv_op(f, args...)` which is overloaded to compute the inverse of an operation, so for example `inv_op(prime, A, 2) = prime(A, -2)`. This is now used in the pullbacks for priming and tagging functions instead of `setinds`.